csrc/scheduler/utils.h

```diff
@@ -42,12 +42,24 @@ constexpr int64_t y_grid_limit = 65535;
 constexpr int64_t z_grid_limit = 65535;
 constexpr int64_t z_block_limit = 64;
 
+// Find largest power of 2 that is a factor of n. If n==0, return largest power
+// of 2 representable by int64_t
 constexpr int64_t maxVectorizationWidth(int64_t n) {
-  int64_t next_vector_size = 2;
-  while (next_vector_size <= n && n % next_vector_size == 0) {
-    next_vector_size <<= 1;
+  if (n == 0) {
+    // Max representable int has null sign bit then all ones. Shift right then
+    // xor to preserve only the most significant bit.
+    int64_t m = std::numeric_limits<int64_t>::max();
+    return m ^ (m >> 1);
   }
-  return next_vector_size >> 1;
+  // For example
+  //   n               = b101101000
+  //           n - 1   = b101100111
+  //        ~ (n - 1)  = b010011000
+  //   n & (~ (n - 1)) = b000001000
+  // The key is that subtracting one flips all trailing 0s as well as the least
+  // significant 1, so all of the other bits will fail the &, leaving
+  // only that 1.
+  return n & (~(n - 1));
```

Comment:
It's not strictly necessary to do this cleanup, but the original code was returning 0 when given a size equal to zero. When using this on a relative address, that can be problematic as it was causing us to set supported vectorization for some operands to 0.